src/datastore/pg-store-v2.ts

```diff
@@ -654,10 +694,50 @@ export class PgStoreV2 extends BasePgStoreModule {
       if (cycleCheck.count === 0)
         throw new InvalidRequestError(`PoX cycle not found`, InvalidRequestErrorType.invalid_param);
       const results = await sql<DbPoxCycleSigner[]>`
-        SELECT
-          signing_key, weight, stacked_amount, weight_percent, stacked_amount_percent
-        FROM pox_sets
-        WHERE canonical = TRUE AND cycle_number = ${args.cycle_number} AND signing_key = ${args.signer_key}
+        WITH signer_keys AS (
+            SELECT DISTINCT ON (stacker) stacker, signer_key
+            FROM pox4_events
+            WHERE canonical = true AND microblock_canonical = true
+                AND name in ('stack-aggregation-commit-indexed', 'stack-aggregation-commit')
+                AND start_cycle_id = ${args.cycle_number}
+                AND end_cycle_id = ${args.cycle_number + 1}
+            ORDER BY stacker, block_height DESC, tx_index DESC, event_index DESC
+        ), delegated_stackers AS (
+            SELECT DISTINCT ON (main.stacker) 
+                main.stacker, 
+                sk.signer_key
+            FROM pox4_events main
+            LEFT JOIN signer_keys sk ON main.delegator = sk.stacker
+            WHERE main.canonical = true 
+                AND main.microblock_canonical = true
+                AND main.name IN ('delegate-stack-stx', 'delegate-stack-increase', 'delegate-stack-extend')
+                AND main.start_cycle_id <= ${args.cycle_number} 
+                AND main.end_cycle_id >= ${args.cycle_number}
+            ORDER BY main.stacker, main.block_height DESC, main.microblock_sequence DESC, main.tx_index DESC, main.event_index DESC
+        ), solo_stackers AS (
+            SELECT DISTINCT ON (stacker) 
+                stacker, 
+                signer_key
+            FROM pox4_events
+            WHERE canonical = true AND microblock_canonical = true
+                AND name in ('stack-stx', 'stacks-increase', 'stack-extend')
```

Comment:
Comment for `stack-extend`, but I think it's ok as is. 

`stack-extend` can be used to change the signing-key, so in some cases a `DISTINCT` might strip one of the keys in a result.
Here, I believe we should only have one valid key/event, since the extend can't override a valid event, but would only append a later cycle entry. But might be trickier for a potential future range query.

Reply:
Ah got it, thanks for the head up